### PR TITLE
Privacy Preserving Analytics Settings (EXPOSUREAPP-4816)

### PR DESCRIPTION
**Pull request by axelherbstreith**
This PR adds a new settings entry for the `Privacy-Preserving Analytics`. When clicking on it, a details screen opens which allows to enable/disable the analytics as well as  change the region etc. In addition, legal texts are shown.

**For testing**, open the settings from the Home Screen and open the newly added entry.

_Short notice: The switch actually does not change any data, this will be done in a separate PR by @kolyaopahle
Further, there will be some polishing in a separate PR as well e.g. aligning the icon size in the settings entry_

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/settings/start/SettingsFragmentViewModel.kt [hunk 1]
```diff
@@ -38,11 +39,20 @@ class SettingsFragmentViewModel @AssistedInject constructor(
         )
     }.asLiveData(dispatcherProvider.Default)
 
+    val data: Boolean = true
```
Comment (kolyaopahle) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/settings/start/SettingsFragmentViewModel.kt:42]: Unused

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/settings/start/SettingsPrivacyPreservingAnalyticsState.kt [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import androidx.annotation.ColorInt
+import de.rki.coronawarnapp.R
+import de.rki.coronawarnapp.util.ContextExtensions.getColorCompat
+import de.rki.coronawarnapp.util.ContextExtensions.getDrawableCompat
+
+data class SettingsPrivacyPreservingAnalyticsState(
+    val isEnabled: Boolean
+) {
+
+    /**
```
Comment (kolyaopahle) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/settings/start/SettingsPrivacyPreservingAnalyticsState.kt:14]: Wrong doc string

**Review by kolyaopahle — APPROVED** [review]
👍 Looks good to me, two small comments that can be adressed in the next pr

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/settings/analytics/SettingsPrivacyPreservingAnalyticsFragment.kt [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    private val vm: SettingsPrivacyPreservingAnalyticsViewModel by cwaViewModels { viewModelFactory }
+    private val binding: FragmentSettingsPrivacyPreservingAnalyticsBinding by viewBindingLazy()
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        binding.apply {
+
+            settingsPpaHeader.headerButtonBack.buttonIcon.setOnClickListener {
+                (activity as MainActivity).goBack()
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/settings/analytics/SettingsPrivacyPreservingAnalyticsFragment.kt:36]: I think here ,you can use `popBackStack()` or `activity.onBackPressed()`

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/settings/analytics/SettingsPrivacyPreservingAnalyticsFragment.kt [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModelFactoryProvider
+import de.rki.coronawarnapp.util.viewmodel.cwaViewModels
+import javax.inject.Inject
+
+class SettingsPrivacyPreservingAnalyticsFragment :
+    Fragment(R.layout.fragment_settings_privacy_preserving_analytics),
+    AutoInject {
+
+    @Inject lateinit var viewModelFactory: CWAViewModelFactoryProvider.Factory
+    private val vm: SettingsPrivacyPreservingAnalyticsViewModel by cwaViewModels { viewModelFactory }
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/settings/analytics/SettingsPrivacyPreservingAnalyticsFragment.kt:27]: Please rename it to `viewModel`

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/settings/analytics/SettingsPrivacyPreservingAnalyticsFragment.kt [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+        vm.ageGroup.observe2(this) {
+            binding.ageGroupRowBody.text = getString(it.labelStringRes)
+        }
+
+        vm.federalState.observe2(this) {
+            binding.districtRow.visibility = if (it != PpaData.PPAFederalState.FEDERAL_STATE_UNSPECIFIED) {
+                View.VISIBLE
+            } else {
+                View.GONE
+            }
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/settings/analytics/SettingsPrivacyPreservingAnalyticsFragment.kt:82]: binding.districtRow.isVisible = it != PpaData.PPAFederalState.FEDERAL_STATE_UNSPECIFIED

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/settings/start/SettingsPrivacyPreservingAnalyticsState.kt [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+     * Formats the settings icon color for background priority
+     */
+    @ColorInt
+    fun getPrivacyPreservingAnalyticsIconColor(c: Context): Int = c.getColorCompat(
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/settings/start/SettingsPrivacyPreservingAnalyticsState.kt:18]: Please rename `c` to `context`

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/settings/start/SettingsPrivacyPreservingAnalyticsState.kt [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+     */
+    fun getPrivacyPreservingAnalyticsIcon(context: Context): Drawable? = context.getDrawableCompat(
+        if (isEnabled) R.drawable.ic_settings_privacy_preserving_analytics_enabled
+        else R.drawable.ic_settings_privacy_preserving_analytics_disabled
+    )
+
+    /**
+     * Formats the text display of settings item status depending on flag provided
+     */
+    fun getPrivacyPreservingAnalyticsText(c: Context): String = c.getString(
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/settings/start/SettingsPrivacyPreservingAnalyticsState.kt:34]: Please rename c to context

File: Corona-Warn-App/src/main/res/layout/fragment_settings.xml [hunk 8]
```diff
@@ -93,14 +98,32 @@
...
+                    app:layout_constraintEnd_toEndOf="parent"
+                    app:layout_constraintStart_toStartOf="parent"
+                    app:layout_constraintTop_toBottomOf="@+id/settings_background_priority"
+                    app:showDivider="@{true}"
+                    app:statusText="@{analyticsState.getPrivacyPreservingAnalyticsText(context)}"
+                    app:subtitle="@{@string/settings_privacy_preserving_analytics_title}"
+                    />
+
+
+
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/res/layout/fragment_settings.xml:118]: two extra lines?

File: Corona-Warn-App/src/main/res/layout/fragment_settings_privacy_preserving_analytics.xml [hunk 9]
```diff
@@ -0,0 +1,430 @@
+<?xml version="1.0" encoding="utf-8"?>
+<layout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto">
+
+
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/res/layout/fragment_settings_privacy_preserving_analytics.xml:5]: extra line

File: Corona-Warn-App/src/main/res/layout/fragment_settings_privacy_preserving_analytics.xml [hunk 10]
```diff
@@ -0,0 +1,430 @@
...
+        <androidx.constraintlayout.widget.Guideline
+            android:id="@+id/guideline_end"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:orientation="vertical"
+            app:layout_constraintGuide_end="@dimen/guideline_end" />
+
+    </androidx.constraintlayout.widget.ConstraintLayout>
+</layout>
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/res/layout/fragment_settings_privacy_preserving_analytics.xml:430]: General comment for the file. when you are referring to ids it is better to use `@id` instead `@+id`
such in `app:layout_constraintTop_toBottomOf="@+id/more_info_row"`
it should be `app:layout_constraintTop_toBottomOf="@id/more_info_row"`

File: Corona-Warn-App/src/main/res/navigation/nav_graph.xml [hunk 11]
```diff
@@ -556,4 +559,12 @@
         android:label="PpaMoreInfoFragment"
         tools:layout="@layout/fragment_ppa_more_info">
     </fragment>
+    <fragment
+        android:id="@+id/settingsPrivacyPreservingAnalyticsFragment"
+        android:name="de.rki.coronawarnapp.ui.settings.analytics.SettingsPrivacyPreservingAnalyticsFragment"
+        android:label="SettingsPrivacyPreservingAnalyticsFragment" >
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/res/navigation/nav_graph.xml]: Please add `tools:layout` it is helpful for other see an overview in the graph

File: Corona-Warn-App/src/main/res/layout/fragment_settings_privacy_preserving_analytics.xml [hunk 12]
```diff
@@ -0,0 +1,430 @@
...
+                        android:layout_height="wrap_content"
+                        android:layout_marginStart="13dp"
+                        android:layout_marginTop="23dp"
+                        android:focusable="true"
+                        android:text="@string/ppa_onboarding_privacy_information_point_sixteen"
+                        app:layout_constraintEnd_toEndOf="parent"
+                        app:layout_constraintStart_toStartOf="@id/bullet_point_consent"
+                        app:layout_constraintTop_toBottomOf="@+id/legal_point_identity" />
+
+                    <ImageView
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/res/layout/fragment_settings_privacy_preserving_analytics.xml:335]: `AppCompatImageView` ?

File: Corona-Warn-App/src/main/res/layout/fragment_settings_privacy_preserving_analytics.xml [hunk 13]
```diff
@@ -0,0 +1,430 @@
...
+                        android:id="@+id/bullet_point_sixteen"
+                        android:layout_width="@dimen/bullet_point_size"
+                        android:layout_height="@dimen/bullet_point_size"
+                        android:layout_marginTop="@dimen/spacing_tiny"
+                        android:importantForAccessibility="no"
+                        android:src="@drawable/bullet_point"
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/res/layout/fragment_settings_privacy_preserving_analytics.xml:341]: `app:srcCompat`

File: Corona-Warn-App/src/main/res/layout/fragment_settings_privacy_preserving_analytics.xml [hunk 14]
```diff
@@ -0,0 +1,430 @@
...
+                        android:layout_height="wrap_content"
+                        android:layout_marginStart="13dp"
+                        android:layout_marginTop="23dp"
+                        android:focusable="true"
+                        android:text="@string/ppa_onboarding_privacy_information_point_identity"
+                        app:layout_constraintEnd_toEndOf="parent"
+                        app:layout_constraintStart_toStartOf="@id/bullet_point_consent"
+                        app:layout_constraintTop_toBottomOf="@+id/legal_point_consent" />
+
+                    <ImageView
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/res/layout/fragment_settings_privacy_preserving_analytics.xml:312]: AppCompatImageView ?

File: Corona-Warn-App/src/main/res/layout/fragment_settings_privacy_preserving_analytics.xml [hunk 15]
```diff
@@ -0,0 +1,430 @@
...
+                        android:id="@+id/bullet_point_identity"
+                        android:layout_width="@dimen/bullet_point_size"
+                        android:layout_height="@dimen/bullet_point_size"
+                        android:layout_marginTop="@dimen/spacing_tiny"
+                        android:importantForAccessibility="no"
+                        android:src="@drawable/bullet_point"
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/res/layout/fragment_settings_privacy_preserving_analytics.xml:318]: `app:srcCompat`

File: Corona-Warn-App/src/main/res/layout/fragment_settings_privacy_preserving_analytics.xml [hunk 16]
```diff
@@ -0,0 +1,430 @@
...
+                        android:layout_height="wrap_content"
+                        android:layout_marginStart="13dp"
+                        android:layout_marginTop="23dp"
+                        android:focusable="true"
+                        android:text="@string/ppa_onboarding_privacy_information_point_consent"
+                        app:layout_constraintEnd_toEndOf="parent"
+                        app:layout_constraintStart_toStartOf="@id/bullet_point_consent"
+                        app:layout_constraintTop_toBottomOf="@+id/legal_body" />
+
+                    <ImageView
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/res/layout/fragment_settings_privacy_preserving_analytics.xml:289]: AppCompatImageView ?

File: Corona-Warn-App/src/main/res/layout/fragment_settings_privacy_preserving_analytics.xml [hunk 17]
```diff
@@ -0,0 +1,430 @@
...
+                        android:id="@+id/bullet_point_consent"
+                        android:layout_width="@dimen/bullet_point_size"
+                        android:layout_height="@dimen/bullet_point_size"
+                        android:layout_marginTop="@dimen/spacing_tiny"
+                        android:importantForAccessibility="no"
+                        android:src="@drawable/bullet_point"
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/res/layout/fragment_settings_privacy_preserving_analytics.xml:295]: `app:srcCompat`

**Comment by mtwalli** [comment]
Few things in functionality

device-2021-02-10-180126
device-2021-02-10-180143
device-2021-02-10-180258

**Comment by axelherbstreith** [comment]
@mtwalli  I will enable the link tomorrow, for the other parts @kolyaopahle will do this in a separate PR

**Review by mtwalli — APPROVED** [review]
Thanks for addressing this so quickly!

**Pull request merged**